0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+
+=== Standard _hold invoice_ process
+
+____
+Assumption: Alice wants to pay Bob _X_ Bitcoin using the LN but Bob does not generate the preimage - Alice does it.
+____
+
+1) Alice generates a secret preimage `p` and sends the hash of it `h(p)` to Bob.
+2) Bob calls `addholdinvoice` with an amount `X` and the hashed preimage `h(p)` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[more details]).
+3) Bob sends the generated invoice to Alice.
+4) Alice now can pay the invoice using `sendpayment`.
+5) As soon as Alice is happy (because she is in a good mood), she tells Bob the secret (or he learns it otherwise, see below).
+6) Bob settles the payment by calling `settleInvoice` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L53[more details]).
+
+The payment is complete.
+
+== Research
+
+____
+Assumption: Bob wants to receive _X_ Bitcoin (**A** Asset) on the LN (**alpha** Ledger) for _Y_ Ether (**B** Asset) on Ethereum (**beta** Ledger)
+Alice magically finds out about this _wish_ or using a [COMIT link](fill in some link).
+
+* Alpha Ledger: Lightning Network
+* Alpha Asset: Bitcoin
+* Beta Ledger: Ethereum
+* Beta Asset: Ether
+____
+
+=== Step-by-step description
+
+1. Alice generates a secret preimage `p` and sends a swap request to Bob, this includes (assets with amounts are emitted):
+
+[cols=3*,options=header]
+|===
+| Name
+| JSON Encoding
+| Description
+
+| `alpha_expiry`
+| `u32`
+| The UNIX timestamp of the time-lock for LN (alpha Ledger)
+
+| `beta_expiry`
+| `u32`
+| The UNIX timestamp of the time-lock on the beta HTLC
+
+| `alpha_refund_identity`
+| `α::Identity`
+| This can be empty as the refund identity will be handled by LN.
+
+| `beta_redeem_identity`
+| `β::Identity`
+| The identity on beta that **B** will be transferred to when the beta-HTLC is activated with the correct secret
+
+| `secret_hash`
+| `hex-encoded-bytes`
+| The output of calling `hash_function` on the secret
+
+|===
+
+[start=2]
+2. Bob _accepts_ the request and performs the following step:
+    .. Create a hold invoice using `addholdinvoice` with the amount `X` and the hashed preimage `secret_hash` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[more details]).
+    .. Bob _subscribes_ to the invoice and waits for the payment either using
+      ... `SubscribeSingleInvoice` - this is not available through the CLI but as RPC; or
+      ... `LookupInvoice` - this is available trough the CLI but needs to be polled regularly.
+3. Bob response to Alice's request with the following information:
+
+[cols=3*,options=header]
+|===
+| Name
+| JSON Encoding
+| Description
+
+| `alpha_redeem_identity`
+| `α::Identity`
+| The identity on alpha that **A**, this is the LN `invoice`
```

Comment:
This seems very hacky. An invoice is not an identity. This should be Bob's Lightning node ID so that Alice knows, where to send the money to. She can then construct the invoice herself if that is what LND needs to make a payment.

Reply:
Yes I agree

Reply:
Partially agree :P 
It's true that the naming (`*_*_identity`) does not fit perfectly as it's not _just_ an identity. 
However, the invoice contains Bob's LN node ID. 
The question we should ask ourselves is whether we can return something differently here as well instead of the pair `alpha_redeem_identity` and `beta_refund_identity` because strictly said: even Bob's LN node ID is not a `redeem_identity`.

Reply:
> strictly said: even Bob's LN node ID is not a redeem_identity.

Correct. It should probably not even be there (e.g. empty).